sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithStateInitialStateSuite.scala

```diff
@@ -497,4 +537,85 @@ class TransformWithStateInitialStateSuite extends StateStoreMetricsTest
       )
     }
   }
+
+  test("transformWithStateWithInitialState - correctness test, " +
```

Comment:
can we add a test with different num of partitions for 1st stream and 2nd stream ?